introduction-to-metamask-api/index.md

```diff
@@ -0,0 +1,389 @@
+
+That's a high level overview, but if you want to deep dive into OpenRPC I suggest checking out [ZaneStarr](https://twitter.com/zanecstarr)'s [Introduction to OpenRPC](https://www.youtube.com/watch?v=9UiKQ1zasjE) session from the [API Specification Conference 2022](https://www.youtube.com/playlist?list=PLcx_iGeB-NxgMrrMMPp3LKQmIixm5g3Q_).
+
+## Essential Tools
+
+This section highlight a few tools that can help you when dealing with:
+
+- Testing JSON RPC servers
+- Creating an Open RPC Spec
+- Working with MetaMask API
+- Interacting with MetaMask using Smart Contracts
+
+### Links to Tools
+
+When building with MetaMask a great resource for the injected `window.ethereum` object is our corresponding Ethereum Provider API docs.
+Want to know about all of the `eth_` and `wallet_` methods supported by MetaMask?
+
+### Ethereum Provider
+
+- [The Ethereum Provider](https://docs.metamask.io/guide/ethereum-provider.html "A link to our MetaMask docs that outline the Ethereum provider")
+
+Check the corresponding RPC API docs that have information on methods we call from our JS code like `eth_requestAccounts`, `wallet_addEthereumChain`, or `wallet_switchEthereumChain`. Maybe you need to add a network/chain to MetaMask adn switch your users to that chain, you can find documentation and code to show you how to do that here:
+
+### MetaMask RPC API
+
+- [MetaMask RPC API](https://docs.metamask.io/guide/rpc-api.html#table-of-contents "A link to our MetaMask docs outlining the MetaMask RPC API")
+
+The next link will allow you to connect a wallet and try these RPC API calls in the browser, we plan on merging the playground with a future version of our docs.
+
+### MetaMask API Playground
+
+- [MetaMask API Playground](https://metamask.github.io/api-playground/ "Connect a wallet and test the MetaMask RPC API")
+
+Chainlink maintains a site called chainlist.org, and it’s a great resource for network/chain information. I often browse the site just looking for the various Ethereum networks and testnets, their chainIds, and the other information needed to connect to them. There are other sources for this information as well, but this is a good hack for quickly finding that information and RPC Server Addresses for each chain with information about their latency and privacy. 
+
+You can add and switch to any chain in the list when MetaMask confirmation pops up you can see what information is needed to add that chain to MetaMask.
+
+### Chainlist
+
+- [Chainlist](https://chainlist.org "Find Ethereum Networks and Chains and add to MetaMask")
+
+Below is an example using Chainlist, I can see Fantom Opera (mainnet), it's chainId and I can add to MetaMask and in the confirmation see the information used to add the network to my wallet.
+
+![](./images//15-chainlist.png)
+
+### Eserialize
+
+- [eserialize](https://eserialize.com/ "Serialize and deserialize the Ethereum Stack")
```

Comment:
`eserialize` => `deserialize` ?

Reply:
Eserialize is just the name of the webtool, is there anything I am missing?

Reply:
Read quickly and thought this was concerning deserialization.